Lib/importlib/_bootstrap_external.py

```diff
@@ -802,7 +802,7 @@ def __init__(self, fullname, path):
         """Cache the module name and the path to the file found by the
         finder."""
         self.name = fullname
-        self.path = path
+        self.path = path.__fspath__() if hasattr(path, '__fspath__') else path
```

Comment:
@lulouie can you point me at the tests that test for `None` and lists? And if accepting `None` is acceptable then you simply test for that case, e.g. `os.fspath(path) if path is not None else None` or something.